src/app/pagination/pagination.component.view.spec.ts

```diff
@@ -0,0 +1,333 @@
+        expect(paginationPageButtonArray[1].querySelector('a').innerHTML).toBe('1');
+        expect(paginationPageButtonArray[2].querySelector('a').innerHTML).toBe('2');
+        expect(paginationPageButtonArray[3].querySelector('i').innerHTML).toBe('chevron_right');
+      });
+    }));
+
+    it('should display page 1 as active', async(() => {
+
+      buildComponent(`
+        <mz-pagination class="col s12 m4"
+          [itemsPerPage]="10"
+          [totalItems]="20"
+        >
+        </mz-pagination>
+      `).then((fixture) => {
+        nativeElement = fixture.nativeElement;
+        fixture.detectChanges();
+
+        expect(paginationPageButtons()[1].classList).toContain('active');
+      });
+    }));
+
+    it('should display page 2 as active', async(() => {
+
+      buildComponent(`
+        <mz-pagination class="col s12 m4"
+          [currentPage]="2"
+          [itemsPerPage]="10"
+          [totalItems]="20"
+        >
+        </mz-pagination>
+      `).then((fixture) => {
+        nativeElement = fixture.nativeElement;
+        fixture.detectChanges();
+
+        expect(paginationPageButtons()[2].classList).toContain('active');
+      });
+    }));
+
+    it('should disable previous page button when first page is active', async(() => {
+
+      buildComponent(`
+        <mz-pagination class="col s12 m4"
+          [itemsPerPage]="10"
+          [totalItems]="20"
+        >
+        </mz-pagination>
+      `).then((fixture) => {
+        nativeElement = fixture.nativeElement;
+        fixture.detectChanges();
+
+        expect(paginationPageButtons()[0].classList).toContain('disabled');
+      });
+    }));
+
+    it('should active page 2 when page 2 is clicked', async(() => {
+
+      buildComponent(`
+        <mz-pagination class="col s12 m4"
+          [itemsPerPage]="10"
+          [totalItems]="20"
+        >
+        </mz-pagination>
+      `).then((fixture) => {
+        nativeElement = fixture.nativeElement;
+        fixture.detectChanges();
+
+        paginationPageButtons()[2].querySelector('a').click();
+        fixture.detectChanges();
+
+        expect(paginationPageButtons()[2].classList).toContain('active');
+      });
+    }));
+
+    it('should active first page when first button is clicked', async(() => {
+
+      buildComponent(`
+        <mz-pagination class="col s12 m4"
+          [itemsPerPage]="10"
+          [totalItems]="20"
+        >
+        </mz-pagination>
+      `).then((fixture) => {
+        nativeElement = fixture.nativeElement;
+        fixture.detectChanges();
+
+        paginationPageButtons()[3].querySelector('a').click();
+        fixture.detectChanges();
+
+        paginationPageButtons()[0].querySelector('a').click();
+        fixture.detectChanges();
+
+        expect(paginationPageButtons()[1].classList).toContain('active');
+      });
+    }));
+
+    it('should active last page when last button is clicked', async(() => {
```

Comment:
This test never pass in the `lastPage()` method of the component because you did not added `[enableFirstAndLastPageButtons]="true"` on the component to show the last page button ... so it is not clicking on the right button